### ISSUE TITLE
Ignoring exact versions

### ISSUE THREAD

**Issue by marceloverdijk** [raineorshine/npm-check-updates#1188]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14`

---

## Steps to Reproduce

```
  "dependencies": {
    "@prisma/client": "^4.3.1",
    "bootstrap": "5.1.3",
    "bootstrap-icons": "^1.9.1",
    "bootstrap-table": "^1.21.0",
    "flag-icons": "^6.6.6",
    "jarallax": "^2.0.4",
    "luxon": "^3.0.3",
    "rellax": "^1.12.1",
    "simplebar": "^5.3.8",
    "smooth-scroll": "^16.1.3",
    "tiny-slider": "^2.9.4"
  },
```

Note above has an exact version from `bootstrap` which I do not want to upgrade when running `ncu -u`.

## Current Behavior

Running `ncu -u` will update `bootstrap` from `5.1.3.` to `5.2.x`.

## Expected Behavior

All other dependencies then `bootstrap` to be updated (if there is an update available), except `bootstrap` as it is defined as an exact version.

**Comment by yanickrochon** [comment]
Funny. I have the _exact_ same question this morning! We have a package that should not get updated because it relies on external resources that can't get updated, thus must remain stable. However `ncu` ignores that the packages is an exact version, thus should not get updated.

**Comment by marceloverdijk** [comment]
Indeed seems quite a reasonable use case ; almost thought I was the only one 😄

**Comment by raineorshine** [comment]
If you'd like npm-check-updates to only upgrade versions that start with `^`, use the following: 

```
ncu --filterVersion "^*"`
```

You can also add `"filterVersion": "^*"` to your .ncurc config file to avoid typing it on the command line.

You may be interested in #1054, which tracks the upcoming `--semver` option. I'm not actively working on it myself, but I think it would be a useful addition.

--- 

Some back story...

If you're wondering "Why isn't this the default behavior?" it has to do with the main purpose of npm-check-updates. From the first line of the README:

> npm-check-updates upgrades your package.json dependencies to the latest versions, ignoring specified versions.

In other words, npm-check-updates was built to *break* your existing version range, while preserving the `^` or `~`, etc. It is just as happy to break an exact version as it is to break the upper bound of a version range (which is what `^` specifies). 

This functionality is *not* available in npm itself. In contrast, upgrading dependencies while respecting semver *is* handled by npm itself: simply run `npm update`. That said, lots of people prefer using exact dependencies and maintaining more control over when and how they are upgraded. npm-check-updates has grown to accommodate the variety of use cases, while maintaining its original *raison d'être*.

**Comment by marceloverdijk** [comment]
Thanks for clarifying @raineorshine this is really helpful.

I ended up adding this to my `package.json`:

```
  "scripts": {
    ..
    "check-updates": "ncu",
    "check-updates:upgrade": "ncu -u --filterVersion '^*'"
  },
```

`check-updates` just informs me of updates (including exact ones) and `check-updates:upgrade` upgrades the deps except the exact ones. Best of both worlds ;-)

**Comment by DarthSonic** [comment]
But what if my version is ">= 5.1.0 <=5.1.3" ? Then the filterVersion will not work and still the check will recommend to update to 5.2.2 currently. There should be an option that a fixed version will never get updated at all, without using any filter.
